topics/admin/tutorials/interactive-tools/tutorial.md

```diff
@@ -893,8 +894,186 @@ Once the playbook run is complete and your Galaxy server has restarted, run the
+>    > psql (10.12 (Ubuntu 10.12-0ubuntu0.18.04.1))
+>    > Type "help" for help.
+>    >
+>    > postgres=#
+>    > ```
+>    {: .code-out}
+>
+> 2. Create a `gxitproxy` database to store the Interactive Tools Session Map.
+>
+>    > <code-in-title>SQL</code-in-title>
+>    > ```sql
+>    > CREATE DATABASE gxitproxy;
+>    > ```
+>    > {: data-cmd="true"}
+>    {: .code-in}
+>
+>    > <code-out-title>SQL</code-out-title>
+>    > ```
+>    > CREATE DATABASE
+>    > ```
+>    {: .code-out}
+>
+> 3. For simplicity, the same user that operates on the Galaxy main database, typically named `galaxy`, is also going to operate on this one. Make this user the owner of the new database.
+>
+>    > <code-in-title>SQL</code-in-title>
+>    > ```sql
+>    > ALTER DATABASE gxitproxy OWNER TO galaxy;
+>    > ```
+>    > {: data-cmd="true"}
+>    {: .code-in}
+>
+>    > <code-out-title>SQL</code-out-title>
+>    > ```
+>    > ALTER DATABASE
+>    > ```
+>    {: .code-out}
+>
+> 4. Sign out of the `postgres` database using `exit`. Then connect to the `gxitproxy` database as `galaxy`.
+>
+>    > <code-in-title>SQL</code-in-title>
+>    > ```sql
+>    > exit
+>    > ```
+>    > {: data-cmd="true"}
+>    {: .code-in}
+>
+>    > <code-in-title>Bash</code-in-title>
+>    > ```bash
+>    > sudo -iu galaxy psql -d gxitproxy
+>    > ```
+>    > {: data-cmd="true"}
+>    {: .code-in}
+>
+>    > <code-out-title>SQL</code-out-title>
+>    > ```
+>    > psql (10.12 (Ubuntu 10.12-0ubuntu0.18.04.1))
+>    > Type "help" for help.
+>    >
+>    > gxitproxy=#
+>    > ```
+>    {: .code-out}
+>
+> 5. Create a `gxitproxy` table in the new database.
+>
+>    > <code-in-title>SQL</code-in-title>
+>    > ```sql
+>    > CREATE TABLE IF NOT EXISTS gxitproxy (key TEXT, key_type TEXT, token TEXT, host TEXT, port INTEGER, info TEXT, PRIMARY KEY (key, key_type));
+>    > ```
+>    > {: data-cmd="true"}
+>    {: .code-in}
+>
+>    > <code-out-title>SQL</code-out-title>
+>    > ```
+>    > CREATE TABLE
+>    > ```
+>    {: .code-out}
+>
+> This is enough to let Galaxy and the Interactive Tool Proxy store the Interactive Tools Session Map in PostgreSQL. But there is a catch: when the Interactive Tool Proxy uses SQLite, it knows the database has changed because it watches the file for changes. When using Postgres, this mechanism is not available. By default, the proxy simply polls the database at regular intervals. To let the user access interactive tools as fast as possible, the proxy can also be notified of updates via [PostgreSQL asynchronous notifications](https://www.postgresql.org/docs/16/libpq-notify.html). To enable them, you have to create a PostgreSQL trigger that sends a NOTIFY message to the channel `gxitproxy` every time the table `gxitproxy` changes.
+>
+> <br>
+> 
+> {:start="6"}
+> 6. Run the following commands to create to create a function that sends a NOTIFY message to the channel `gxitproxy` and a trigger that runs the function every time the table `gxitproxy` changes.
```

Comment:
```suggestion
> 6. This is enough to let Galaxy and the Interactive Tool Proxy store the Interactive Tools Session Map in PostgreSQL. But there is a catch: when the Interactive Tool Proxy uses SQLite, it knows the database has changed because it watches the file for changes. When using Postgres, this mechanism is not available. By default, the proxy simply polls the database at regular intervals. To let the user access interactive tools as fast as possible, the proxy can also be notified of updates via [PostgreSQL asynchronous notifications](https://www.postgresql.org/docs/16/libpq-notify.html). To enable them, you have to create a PostgreSQL trigger that sends a NOTIFY message to the channel `gxitproxy` every time the table `gxitproxy` changes.
>
>    Run the following commands to create to create a function that sends a NOTIFY message to the channel `gxitproxy` and a trigger that runs the function every time the table `gxitproxy` changes.
```